src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -35174,97 +35497,80 @@ class C
+                    static List<int> M1(MyCollection<int> c) => [..c];
+
+                    #nullable enable
+                    static List<object> M2(MyCollection<object> c) => [..c];
+                    static List<object?> M3(MyCollection<object> c) => [..c];
                 }
                 """;
 
-            var expectedOutputString = "[1, 2, 3],";
-            var verifier = CompileAndVerify(
-                [source, s_collectionExtensions],
-                expectedOutput: targetFramework == TargetFramework.Standard ? expectedOutputString : IncludeExpectedOutput(expectedOutputString),
-                targetFramework: targetFramework,
-                verify: Verification.Skipped);
-
+            var verifier = CompileAndVerify([source, s_collectionExtensions], expectedOutput: "[1, 2, 3], [1, 2, 3], [1, 2, 3], ", verify: Verification.Skipped);
             verifier.VerifyDiagnostics();
 
-            var expectedIL = targetFramework == TargetFramework.Standard ? """
+            verifier.VerifyIL("C.M1", """
                 {
-                  // Code size       21 (0x15)
-                  .maxstack  3
-                  .locals init (MyCollection V_0)
+                  // Code size        7 (0x7)
+                  .maxstack  1
                   IL_0000:  ldarg.0
-                  IL_0001:  stloc.0
-                  IL_0002:  ldloc.0
-                  IL_0003:  callvirt   "int MyCollection.Count.get"
-                  IL_0008:  newobj     "System.Collections.Generic.List<int>..ctor(int)"
-                  IL_000d:  dup
-                  IL_000e:  ldloc.0
-                  IL_000f:  callvirt   "void System.Collections.Generic.List<int>.AddRange(System.Collections.Generic.IEnumerable<int>)"
-                  IL_0014:  ret
+                  IL_0001:  call       "System.Collections.Generic.List<int> System.Linq.Enumerable.ToList<int>(System.Collections.Generic.IEnumerable<int>)"
+                  IL_0006:  ret
                 }
-                """ : """
+                """);
+            verifier.VerifyIL("C.M2", """
                 {
-                  // Code size       91 (0x5b)
-                  .maxstack  3
-                  .locals init (int V_0,
-                                System.Collections.Generic.List<int> V_1,
-                                System.Span<int> V_2,
-                                int V_3,
-                                MyCollection.Enumerator V_4,
-                                int V_5)
+                  // Code size        7 (0x7)
+                  .maxstack  1
+                  IL_0000:  ldarg.0
+                  IL_0001:  call       "System.Collections.Generic.List<object> System.Linq.Enumerable.ToList<object>(System.Collections.Generic.IEnumerable<object>)"
+                  IL_0006:  ret
+                }
+                """);
+            // This should be the same as M2. See https://github.com/dotnet/roslyn/issues/74894
```

Comment:
>https://github.com/dotnet/roslyn/issues/74894 [](http://example.com/codeflow?start=50&length=45)

Should we fix this issue before merging this PR?

Reply:
Do you mean to fix it in a separeate PR or right here? Since we already moved this peace of logic and touched related lines, there are inevitable merge conflicts if you think of a separate PR.

Reply:
I've created a separate PR: https://github.com/dotnet/roslyn/pull/74949

Hopefully, the merge conflicts with this PR should be relatively small.